content/blog/2022/2022-08-15-admin-operations-with-nodejs-launchdarkly.md

```diff
@@ -0,0 +1,681 @@
+
+To implement this, we will use a third-party library, [Express Rate Limit](https://www.npmjs.com/package/express-rate-limit). So, to install it, run:
+
+```bash
+npm install express-rate-limit
+```
+
+First, we will define a simple express app to host a server:
+
+```javascript
+import bodyParser from 'body-parser';
+import express from 'express';
+import cors from 'cors';
+import rateLimit from 'express-rate-limit';
+import LaunchDarkly from 'launchdarkly-node-server-sdk';
+import LdLogger from './ld_logger.js';
+
+// Initiating LaunchDarkly Client
+const LD_SDK_KEY = 'sdk-d2432dc7-e56a-458b-9f93-0361af47d578';
+const userName = 'admin';
+const launchDarklyClient = LaunchDarkly.init( LD_SDK_KEY );
+
+// Initiating the Logger
+const flagKey = 'backend-log-level';
+let logger;
+launchDarklyClient.once('ready', async () => {
+    logger = new LdLogger( launchDarklyClient, flagKey, userName );
+    serverInit();
+  }
+);
+
+const serverInit = async () => {
+    // Essential globals
+    const app = express();
+
+    // Initialize API
+    app.get('/hello', function (req, res) {
+        return res.send('Hello World')
+    });
+
+    // Initialize server
+    app.listen(5000, () => {
+        logger.info('Starting server on  port 5000');
+    });
+};
+```
+
+Now we will look at the JSON type flags as these feature flag values are pretty open-ended. LaunchDarkly supports a JSON type right out of the box. This allows us to pass Object and Array data structures to our application which can then be used to implement lightweight administrative and operational functionality in our web application. In this case, we will define a feature flag that will take the sliding window counter config and use it in our middleware.
+
+{{% image alt="Rate Limiter Flag" src="images/posts/nodejs-backend-launchdarkly/Rate_Limiter_Flag.png" %}}
+
+{{% image alt="Rate Limiter Variation" src="images/posts/nodejs-backend-launchdarkly/Rate_Limiter_Variation.png" %}}
+
+Next we will define a middleware and pass it to the `express` app before starting the server:
+
+```javascript
+// Initialize Rate Limit Midlleware
+const rateLimiterConfig = await launchDarklyClient.variation(
+    'rate-limiter-config',
+    {
+        key: userName // The static "user" for this task.
+    },
+    {
+        windowMs: 24 * 60 * 60 * 1000, // 24 hrs in milliseconds
+        max: 100, // Limit each IP to 100 requests per `window` (here, per 24 hours).
+        // Set it to 0 to diable rateLimiter
+        message: 'You have exceeded 100 requests in 24 hrs limit!', 
+        standardHeaders: true, // Return rate limit info in the `RateLimit-*` headers
+        legacyHeaders: false, // Disable the `X-RateLimit-*` headers
+     } // Default/fall-back value
+);
+app.use(rateLimit(rateLimiterConfig));
+```
+
+Next we can define the script as part of `package.json`:
+
+```json
+{
+	"scripts": {
+    	"rateLimiter": "nodemon rate_limiter.js"
```

Comment:
why use `nodemon` here and not `node` as in the previous examples? If there's a reason, please explain in the text, if not use `node`

Reply:
this was a typo as I was using it for my dev. But i have replaced it with `node`.